credsweeper/app.py

```diff
@@ -104,25 +108,44 @@ def run(self, content_provider: FilesProvider) -> None:
             content_provider: path objects to scan
 
         """
-        file_extractors = []
-        if content_provider:
-            file_extractors = content_provider.get_scannable_files(self.config)
+        _empty_list_mypy_fix: List[TextContentProvider] = []
+        file_extractors: Union[List[DiffContentProvider], List[TextContentProvider]] = \
+            content_provider.get_scannable_files(self.config) if content_provider else _empty_list_mypy_fix
         logging.info("Start Scanner")
         self.scan(file_extractors)
         self.post_processing()
         self.export_results()
 
-    def scan(self, file_providers: List[ContentProvider]) -> None:
-        """Run scanning of files from an argument "file_providers".
+    def scan(self, content_providers: Union[List[DiffContentProvider], List[TextContentProvider]]) -> None:
+        """Run scanning of files from an argument "content_providers".
 
         Args:
-            file_providers: file objects to scan
+            content_providers: file objects to scan
 
         """
+        if 1 < self.pool_count:
+            self.__multi_jobs_scan(content_providers)
+        else:
+            # one thread flow
+            all_cred: List[Candidate] = []
+            for i in content_providers:
+                candidates = self.file_scan(i)
+                all_cred.extend(candidates)
+            if self.config.api_validation:
+                api_validation = ApplyValidation()
+                for cred in all_cred:
+                    logging.info("Run API Validation")
+                    cred.api_validation = api_validation.validate(cred)
+                    self.credential_manager.add_credential(cred)
+            else:
+                self.credential_manager.set_credentials(all_cred)
```

Comment:
Can we support `single_job_scan` with before code? Not separating `multi_job_scan` and `single_job_scan` like this code.. Please give your opinion about why it is needed.

And It seems `single_job_scan` code block can be extracted to method some like `__single_job_scan()`.

Reply:
multiprocessing creates new process even for single job. It is extra and makes debug complicated.

Reply:
Okay. Then how about extract method for `single_job_scan`? :slightly_smiling_face:

Reply:
https://github.com/Samsung/CredSweeper/pull/151/commits/8a4515db68fe3827fc32a8d390066dc48649780c